net: use `log::error!` to handle DevTools startup failures gracefully

Integral-Tech opened the pull request:
Replace the `unwrap()` call with `log::error!()` in the `send_request_to_devtools()` function to enhance error handling.

---

Before:
```
╰─❯ ./servo --devtools=1
[2025-09-15T14:35:07Z ERROR servoshell::desktop::app_state] Saw Servo error: DevtoolsFailedToStart!
called `Result::unwrap()` on an `Err` value: "SendError(..)" (thread tokio-runtime-5, at components/net/http_loader.rs:435)
[2025-09-15T14:35:07Z ERROR servoshell::panic_hook] called `Result::unwrap()` on an `Err` value: "SendError(..)"
```

After:
```
╰─❯ ./servo --devtools=1
[2025-09-16T01:24:59Z ERROR servoshell::desktop::app_state] Saw Servo error: DevtoolsFailedToStart!
[2025-09-16T01:24:59Z ERROR net::http_loader] DevTools send failed: sending on a disconnected channel
[2025-09-16T01:24:59Z ERROR net::http_loader] DevTools send failed: sending on a disconnected channel
...
```

jdm reviewed (changes requested):
I would prefer not to panic here. If something goes wrong with the devtools, we should be able to keep running the browser and log an error.

Taym95 commented on components/net/http_loader.rs:
```diff
@@ -430,9 +430,9 @@ pub fn send_request_to_devtools(
-        .send(DevtoolsControlMsg::FromChrome(msg))
-        .unwrap();
+    if let Err(e) = devtools_chan.send(DevtoolsControlMsg::FromChrome(msg)) {
+        panic!("{e}");
```
Replacing .unwrap() with panic!() doesn’t make failure handling more “graceful”, both crash the process. What we want here is either:
- propagate the error to the caller, or
- keep running but add good observability (log + context), or
- if crashing is truly intended, use expect() with a clear, actionable message.

in this case crashing is not needed and we would prefer loging:
```
if let Err(e) = devtools_chan.send(DevtoolsControlMsg::FromChrome(msg)) {
        log::error!("DevTools send failed: {e}");
}
```

Integral-Tech committed:
net: use `log::error!` to handle DevTools startup failures gracefully

Replace the `unwrap()` call with `log::error!()` in the
`send_request_to_devtools()` function to enhance error handling.

Signed-off-by: Integral <integral@member.fsf.org>

jdm reviewed (approved):
Thanks!
